### DISCUSSION TITLE
Failed to open the file foundryup-zksync: Permission denied

### DISCUSSION THREAD

**Discussion by KingVics** [Cyfrin/foundry-full-course-cu#3039] [General]
Hi community, I need help install foundry-zksync for windows 11 computer.
I already cloned the foundry Zksync repository, but on running ./install-foundry-zksync from my bash terminal 
I get this error:
Failed to open the file foundryup-zksync: Permission denied
curl: (23) Failure writing output to destination, passed 1378 returned 4294967295

**Comment by EngrPips** [comment]
Hello @KingVics, I asked Gemini Ai and below was what it said 

**1. Check File Permissions**

* **Locate the file:** Find the exact path to the `foundryup-zksync` file within the cloned Zksync repository.
* **Check permissions:** Use the `ls -l` command in your terminal to check the file permissions. For example:
   ```bash
   ls -l path/to/foundryup-zksync 
   ```
* **Adjust permissions (if necessary):** If the file permissions are restrictive, use the `chmod` command to grant execute permissions. For example:
   ```bash
   chmod +x path/to/foundryup-zksync 
   ```

**2. Verify Curl Installation and Configuration**

* **Check Curl installation:** Ensure you have Curl installed on your system. If not, install it using your system's package manager (e.g., `apt-get install curl` on Debian/Ubuntu).
* **Check Curl configuration:** Sometimes, incorrect Curl configurations can cause write issues. Check your Curl configuration file (`/etc/curl/curl.conf` or similar) for any restrictions that might be preventing the file from being written.

**3. Check for Disk Space Issues**

* **Check available disk space:** Ensure you have sufficient free disk space to download and install the necessary files. Use the `df -h` command to check your disk space usage.

**4. Retry the Command**

* **Clear the terminal:** Close the terminal window and open a new one.
* **Re-run the command:** Execute `./install-foundry-zksync` again.

**5. Check for Network Connectivity**

* **Test your internet connection:** Ensure you have a stable internet connection.
* **Check for network firewalls:** Verify that no firewalls or network security rules are blocking the download.

**6. Consult Zksync Documentation**

* **Refer to official documentation:** Check the official Zksync documentation for any specific installation instructions or troubleshooting guides.

**If the issue persists:**

* **Provide more context:** If the above steps don't resolve the issue, provide more details about your environment (operating system, specific Zksync version, etc.).
* **Seek community support:** Reach out to the Zksync community forums or Discord channels for assistance.

I hope this helps! Let me know if you have any further questions or encounter any other issues.

**Reply by EngrPips** [reply]
Please try those suggestions and let me know if any of the suggestions helped.
